doc/classes/PoolByteArray.xml

```diff
@@ -5,7 +5,7 @@
 	</brief_description>
 	<description>
 		An [Array] specifically designed to hold bytes. Optimized for memory usage, does not fragment the memory.
-		[b]Note:[/b] This type is passed by value and not by reference.
+		[b]Note:[/b] This type is usually passed by value and not by reference. However, if a [PoolByteArray] is stored as a class property or as the value of an [Array] or [Dictionary], it will be passed as reference. Due to limitations, attempting to modify a [PoolByteArray] passed as reference with a method such as [method append] won't work.
```

Comment:
I'm a bit confused around this topic but isn't it the other way around? It's because the array is passed by value when indexing a dict that you can't modify the one that is stored inside the dict - you'd need a reference to modify it.